pkg/bootstrap/api/types.go

```diff
@@ -80,6 +86,12 @@ const (
 	// authenticate as. The full username given is "system:bootstrap:<token-id>".
 	BootstrapUserPrefix = "system:bootstrap:"
 
-	// BootstrapGroup is the group bootstrapping bearer tokens authenticate in.
-	BootstrapGroup = "system:bootstrappers"
+	// BootstrapGroupPattern is the valid regex pattern that all groups
+	// assigned to a bootstrap token by BootstrapTokenExtraGroupsKey must match.
+	// See also ValidateBootstrapGroupName().
+	BootstrapGroupPattern = "system:bootstrappers:[a-z0-9:-]{0,255}[a-z0-9]"
```

Comment:
Nit: usually for stuff like this we use a DNS-ish type of regexp.  That means that any segment shouldn't start with a number or -. It is a bit strange to allow `system:bootstrappers:-foo` here, for example.

But I don't see any real problem to be honest so it is just a nit.